### ISSUE TITLE
Port given through config is not used

### ISSUE THREAD

**Issue by berndstelzl** [docker-library/redis#207]
I am currently trying to get a redis cluster running, using docker host networking.

In script, I fire up the redis instances one by one using following command:

`docker run -d -v redis_"$i".conf:/usr/local/etc/redis/redis.conf --network host --name myredis_"$i" redis redis-server /usr/local/etc/redis/redis.conf &`

with $i being the port number this instance uses, so I can easily identify them later on.

My config files look like this:

```
port 63xx
cluster-enabled yes
cluster-config-file nodes.conf
cluster-node-timeout 5000
appendonly yes
```

With xx being the port number in the config, as I have multiple configs.

When starting the instances one by one, each with a different config containing a different port number, I get the following:

```
> $ 1:C 07 Oct 2019 10:14:32.895 # oO0OoO0OoO0Oo Redis is starting oO0OoO0OoO0Oo                                                                                                                           
1:C 07 Oct 2019 10:14:32.895 # Redis version=5.0.6, bits=64, commit=00000000, modified=0, pid=1, just started
1:C 07 Oct 2019 10:14:32.895 # Configuration loaded
1:M 07 Oct 2019 10:14:32.897 # Could not create server TCP listening socket *:6379: bind: Address already in use
```

Basically, no matter the port I hand over using the config file, it always uses the default port.

**Comment by wglambert** [comment]
I can't reproduce
```console
$ docker run -d --name redis --network host -v "$PWD"/redis.conf:/usr/local/etc/redis/redis.conf redis redis-server /usr/local/etc/redis/redis.conf
20a5976e60d5cc89d18c797c23d79bf5efb584b4019224a1ac5bd02b3d29454e

$ docker logs redis -f
1:C 07 Oct 2019 16:45:31.473 # oO0OoO0OoO0Oo Redis is starting oO0OoO0OoO0Oo
1:C 07 Oct 2019 16:45:31.473 # Redis version=5.0.6, bits=64, commit=00000000, modified=0, pid=1, just started                                     
1:C 07 Oct 2019 16:45:31.473 # Configuration loaded                                                                                               
1:M 07 Oct 2019 16:45:31.474 * No cluster configuration found, I'm 6539dfce07704b0bfd020abe24ff500674232fa3                                       
1:M 07 Oct 2019 16:45:31.856 * Running mode=cluster, port=6391.                                                                                   
1:M 07 Oct 2019 16:45:31.856 # WARNING: The TCP backlog setting of 511 cannot be enforced because /proc/sys/net/core/somaxconn is set to the lower value of 128.                                                                                                                                    
1:M 07 Oct 2019 16:45:31.856 # Server initialized                                                                                                 
1:M 07 Oct 2019 16:45:31.856 # WARNING you have Transparent Huge Pages (THP) support enabled in your kernel. This will create latency and memory usage issues with Redis. To fix this issue run the command 'echo never > /sys/kernel/mm/transparent_hugepage/enabled' as root, and add it to your /etc/rc.local in order to retain the setting after a reboot. Redis must be restarted after THP is disabled.                                        
1:M 07 Oct 2019 16:45:31.856 * Ready to accept connections                                                                                        
^C                                                                                                                                                

$ sudo netstat -tulpn | grep 6391
tcp        0      0 0.0.0.0:6391            0.0.0.0:*               LISTEN      13946/redis-server  
tcp        0      0 0.0.0.0:16391           0.0.0.0:*               LISTEN      13946/redis-server  
tcp6       0      0 :::6391                 :::*                    LISTEN      13946/redis-server  
tcp6       0      0 :::16391                :::*                    LISTEN      13946/redis-server
```

Maybe your config wasn't read and it tried to use the default which was taken 
`Could not create server TCP listening socket *:6379: bind: Address already in use`

**Comment by tianon** [comment]
With the command you've given, you've probably created a literal named volume called `redis_$i.conf`, which is created as an empty directory (and Redis isn't going to know what to do with that).  For a bind mount to work, you need to provide a full path from the host (or use a variable like `$PWD` as in @wglambert's example to do so).
